skbio/sequence/_nucleotide_mixin.py

```diff
@@ -161,6 +161,12 @@ def complement(self, reverse=False):
 
         if reverse:
             complement = complement[::-1]
+
+        if self.has_interval_metadata():
+            interval_metadata = self.interval_metadata
+            if reverse:
+                interval_metadata._reverse()
```

Comment:
Add a TODO noting that this can be revised to match `positional_metadata` when `__getitem__` supports `interval_metadata`.